src/lib/y2storage/autoinst_profile/section_with_attributes.rb

```diff
@@ -95,6 +108,23 @@ def to_hashes
         end
       end
 
+      # Returns the section name
+      #
+      # In some cases, the section name does not match with the XML name
+      # and this method should be redefined.
+      #
+      # @example
+      #   section = PartitioningSection.new
+      #   section.section_name #=> "partitioning"
+      #
+      # @return [String] Section name
+      def section_name
+        klass_name = self.class.name.split("::").last
+        klass_name
+          .gsub(/([a-z])([A-Z])/, "\\1_\\2").downcase
+          .gsub(/_section\z/, "")
```

Comment:
gsub here is not needed, it can be simple sub or even better `.chomp("_section")`